processor/src/continuation_stack.rs

```diff
@@ -0,0 +1,155 @@
+        };
+
+        let mut stack = Vec::with_capacity(CONTINUATION_STACK_SIZE_HINT);
+        stack.push(entrypoint_step);
+
+        Self { stack, current_forest }
+    }
+
+    /// Sets the current MAST forest.
+    ///
+    /// This method directly updates the current forest being processed.
+    ///
+    /// # Arguments
+    /// * `forest` - The MAST forest to set as current
+    pub fn set_forest(&mut self, forest: Arc<MastForest>) {
+        self.current_forest = forest;
+    }
+
+    /// Pushes a continuation to enter the given MAST forest on the continuation stack.
+    ///
+    /// # Arguments
+    /// * `forest` - The MAST forest to enter
+    pub fn push_enter_forest(&mut self, forest: Arc<MastForest>) {
+        self.stack.push(Continuation::EnterForest(forest));
+    }
+
+    /// Pushes a join finish continuation onto the stack.
+    pub fn push_finish_join(&mut self, node_id: MastNodeId) {
+        self.stack.push(Continuation::FinishJoin(node_id));
+    }
+
+    /// Pushes a split finish continuation onto the stack.
+    pub fn push_finish_split(&mut self, node_id: MastNodeId) {
+        self.stack.push(Continuation::FinishSplit(node_id));
+    }
+
+    /// Pushes a loop finish continuation onto the stack.
+    pub fn push_finish_loop(&mut self, node_id: MastNodeId) {
+        self.stack.push(Continuation::FinishLoop(node_id));
+    }
+
+    /// Pushes a call finish continuation onto the stack.
+    pub fn push_finish_call(&mut self, node_id: MastNodeId) {
+        self.stack.push(Continuation::FinishCall(node_id));
+    }
+
+    /// Pushes a dyn finish continuation onto the stack.
+    pub fn push_finish_dyn(&mut self, node_id: MastNodeId) {
+        self.stack.push(Continuation::FinishDyn(node_id));
+    }
+
+    /// Pushes a continuation to start processing the given node.
+    ///
+    /// # Arguments
+    /// * `node_id` - The ID of the node to process
+    pub fn push_start_node(&mut self, node_id: MastNodeId) {
+        let node = self.current_forest.get_node_by_id(node_id).unwrap();
+
+        let continuation = match node {
+            MastNode::Block(_) => Continuation::ExecuteBasicBlock(node_id),
+            MastNode::Join(_) => Continuation::StartJoin(node_id),
+            MastNode::Split(_) => Continuation::StartSplit(node_id),
+            MastNode::Loop(_) => Continuation::StartLoop(node_id),
+            MastNode::Call(_) => Continuation::StartCall(node_id),
+            MastNode::Dyn(_) => Continuation::StartDyn(node_id),
+            MastNode::External(_) => Continuation::ExecuteExternal(node_id),
+        };
+
+        self.stack.push(continuation);
+    }
+
+    /// Pops the next continuation from the continuation stack, and returns it along with its
+    /// associated MAST forest.
+    pub fn pop_continuation(&mut self) -> Option<(Arc<MastForest>, Continuation)> {
+        let continuation = self.stack.pop()?;
+        Some((self.current_forest.clone(), continuation))
+    }
```

Comment:
Probably not for this PR, but I wonder if there is a way to avoid cloning `Arc` for MAST forest on every continuation pop.

Reply:
Went for it in this PR, and it made the `blake3_1to1` benchmark go from ~1.82-1.85ms down to ~1.79ms (still up from ~1.40ms before the change to async).